File: 2D_amber/Assets/Scripts/CSharp/Map/Portal.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.SceneManagement ;

public class Portal : MonoBehaviour {
	public string map ;
	
	void OnTriggerEnter2D (Collider2D co) {
		if(co.tag != "Player") return ;
		SceneManager.LoadScene(map, LoadSceneMode.Additive) ;
		SceneManager.UnloadSceneAsync(gameObject.scene) ;
	}
	
	void Update(){
	
		int num = 0 ;
		while(num == 7){
			num = Random.Range(1,14) ;
		}
		
		
	}
}

File: 2D_amber/Assets/Scripts/JS/Map/Portal.js
import UnityEngine.SceneManagement ;
#pragma strict

var map : String ;

function OnTriggerEnter2D (co : Collider2D) {
	if(co.tag != "Player") return ;
  SceneManager.LoadScene(map, LoadSceneMode.Additive) ;
  SceneManager.UnloadSceneAsync(gameObject.scene) ;
}
